File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/crypto/impl/NativeBase64.java
package at.tugraz.tc.cyfile.crypto.impl;

import at.tugraz.tc.cyfile.crypto.Base64;

public class NativeBase64 implements Base64 {
    @Override
    public byte[] decode(String encoded) {
        return android.util.Base64.decode(encoded, android.util.Base64.DEFAULT);
    }

    @Override
    public String encode(byte[] bytes) {
        return android.util.Base64.encodeToString(bytes, android.util.Base64.DEFAULT);
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/crypto/impl/NativeBase64.kt
package at.tugraz.tc.cyfile.crypto.impl

import at.tugraz.tc.cyfile.crypto.Base64

class NativeBase64 : Base64 {
    override fun decode(encoded: String): ByteArray {
        return android.util.Base64.decode(encoded, android.util.Base64.DEFAULT)
    }

    override fun encode(bytes: ByteArray): String {
        return android.util.Base64.encodeToString(bytes, android.util.Base64.DEFAULT)
    }
}
